test/unit/math/prim/arr/err/check_opencl_test.cpp

```diff
@@ -0,0 +1,16 @@
+#ifdef STAN_OPENCL
+#include <stan/math/prim/mat.hpp>
```

Comment:
don't include both. This should only include `arr` since it's in the `arr` test.